enumerable.rb

```diff
@@ -0,0 +1,86 @@
+# rubocop:disable Style/For
+
+# Enumerable module
+module Enumerable
+  def my_each
+    return unless block_given?
+
+    for i in self
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    return unless block_given?
+
+    index = 0
+    for i in self
+      yield i, index
+      index += 1
+    end
+  end
+
+  def my_select
+    return unless block_given?
+
+    array = []
+    my_each { |x| array << x if yield x }
+
+    array
+  end
+
+  def my_all?
+    return unless block_given?
+
+    result = my_select { |x| yield x }
+    length == result.length
+  end
+
+  def my_any?
+    return unless block_given?
+
+    result = my_select { |x| yield x }
+    result.length.positive?
+  end
```

Comment:
- [ ]  when no block or argument is given returns true if at least one of the collection is not false or nil

- [ ] when no block or argument is given returns false if at least one of the collection is not true

- [ ] when a class is passed as an argument returns true if at least one of the collection is a member of such class

- [ ] when a Regex is passed as an argument returns true if any of the collection matches the Regex

- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex

- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

- [ ] when a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern